src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/TrackingOptInModal.java

```diff
@@ -20,153 +20,161 @@
+    setGeoCookie(driver, EU_CONTINENT, country);
+    getUrl(urlWithInstantGlobals(instantGlobals, page));
+    clickRejectButton();
+  }
+
+  public void verifyTrackingRequestsNotSend(List<String> urlPatterns,
+                                            NetworkTrafficInterceptor networkTrafficInterceptor) {
+    isTrackingRequestsNotSend(urlPatterns, networkTrafficInterceptor);
+  }
+
+  public void verifyTrackingRequestsSend(List<String> urlPatterns,
+                                         NetworkTrafficInterceptor networkTrafficInterceptor) {
+    isTrackingRequestSend(urlPatterns, networkTrafficInterceptor);
+  }
+
+  public String urlOptInModalDisplayedOasis(Page page) {
+    return urlBuilder.appendQueryStringToURL(urlBuilder.getUrlForPage(page), MODAL_INSTANT_GLOBAL);
+  }
+
+  public void isTrackingRequestsNotSend(List<String> elementsList,
+                                        NetworkTrafficInterceptor networkTrafficInterceptor) {
+    wait.forX(WAITNG_TIME_FOR_ALL_REQUESTS);
+    for (int i = 0; i < elementsList.size(); i++) {
+      Assertion.assertFalse(
+          isSuccessfulResponseByUrlPattern(networkTrafficInterceptor, elementsList.get(i)),
+          "Request to " + elementsList.get(i) + " services was found");
     }
+  }
 
-    public void verifyTrackingRequestsSend(List<String> urlPatterns,
-                                                 NetworkTrafficInterceptor networkTrafficInterceptor,
-                                                 Page page){
-        getUrl(urlBuilder.appendQueryStringToURL(urlBuilder.getUrlForPage(page), MODAL_INSTANT_GLOBAL));
-        clickAcceptButton();
-        isTrackingRequestSend(urlPatterns, networkTrafficInterceptor);
+  public void isTrackingRequestSend(List<String> elementsList,
+                                    NetworkTrafficInterceptor networkTrafficInterceptor) {
+    for (int i = 0; i < elementsList.size(); i++) {
+      wait.forSuccessfulResponseByUrlPattern(networkTrafficInterceptor, elementsList.get(i));
     }
+  }
 
-    public String urlOptInModalDisplayedOasis(Page page){
-        return urlBuilder.appendQueryStringToURL(urlBuilder.getUrlForPage(page), MODAL_INSTANT_GLOBAL);
-    }
+  private boolean isSuccessfulResponseByUrlPattern(
+      final NetworkTrafficInterceptor trafficInterceptor,
+      final String pattern) {
+    try {
+      entry = trafficInterceptor.getEntryByUrlPattern(pattern);
+      return entry.getResponse().getStatus() < 400;
 
-    public void isTrackingRequestsNotSend(List<String> elementsList,
-                                          NetworkTrafficInterceptor networkTrafficInterceptor) {
-        wait.forX(WAITNG_TIME_FOR_ALL_REQUESTS);
-        for(int i=0; i<elementsList.size(); i++){
-            Assertion.assertFalse(isSuccessfulResponseByUrlPattern(networkTrafficInterceptor, elementsList.get(i)),
-                    "Request to " + elementsList.get(i) + " services was found");
-        }
+    } catch (NullPointerException ex) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)